other/dns-policy-and-dns-config/01-policy.yaml

```diff
@@ -0,0 +1,7 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+apply:
```

Comment:
Guessing from the output that this is already part of a cluster. I think we should still have it, though. For example, in a K3d/K3s cluster, this isn't present since those don't use kubeadm as a bootstrapper.

Reply:
Yes, It is part of a kind cluster by default. But makes sense, I'll add it.

Basically this uses the `kubeadm-config` to just get the cluster names. Is there any way we can get the clustername from just the kubeconfig of the test? I mean I can think of commands to do it, but how do I add it as a policy variable?

Reply:
I would just store the ConfigMap and be done with it.

Reply:
But isn't there a possibility of this messing up with the already existing `kubeadm-config`? Should I store it somewhere and then restore after the test?

Reply:
What does the existing one have in it? Can you show a sample from a KinD cluster?

Reply:
Or I'll just run a `kubectl create -f` and just ignore the error code?

Reply:
```yaml
apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      certSANs:
      - localhost
      - 127.0.0.1
      extraArgs:
        authorization-mode: Node,RBAC
        runtime-config: ""
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta3
    certificatesDir: /etc/kubernetes/pki
    clusterName: kind
    controlPlaneEndpoint: kind-control-plane:6443
    controllerManager:
      extraArgs:
        bind-address: 0.0.0.0
        enable-hostpath-provisioner: "true"
    dns: {}
    etcd:
      local:
        dataDir: /var/lib/etcd
        extraArgs:
          listen-metrics-urls: http://0.0.0.0:2382
    imageRepository: registry.k8s.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.27.2
    networking:
      dnsDomain: cluster.local
      podSubnet: 10.244.0.0/16
      serviceSubnet: 10.96.0.0/16
    scheduler:
      extraArgs:
        bind-address: 0.0.0.0
kind: ConfigMap
metadata:
  creationTimestamp: "2023-06-23T08:19:56Z"
  name: kubeadm-config
  namespace: kube-system
  resourceVersion: "234"
  uid: 7346e11d-916a-4427-b3bb-beeb5ea3e92d```

Reply:
A `kubectl create` would automatically check for the existence and fail if it's already there right? So I can use that?

Reply:
There's a `kubeadm-config` already existing in the policy folder. I'll reuse that and tweak the CLI test a bit.